File: XMLCheck/Handler/CheckRulesXML02/Xml2RuleBase.cs
using XMLCheck.Handler.CheckRulesXML2;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML02
{
    public abstract class Xml2RuleBase:IXml2Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml2Model model, HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml2/xml2RuleBase.ts
import { Xml2Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { BaseRule } from '../base.rule';

export interface IXml2Rule {
  readonly key: string;
  readonly isEnabled: boolean;
  check(model: Xml2Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml2RuleBase extends BaseRule<Xml2Model> implements IXml2Rule {}
